lms/product/plugin/course_copy.py

```diff
@@ -20,7 +20,7 @@ def is_file_in_course(self, course_id, file_id, type_) -> bool:
 
     def find_matching_file_in_course(
         self,
-        store_new_course_files: Callable[[str], None],
+        store_new_course_files: Callable[[str], Any] | Callable[[int], Any],
```

Comment:
`Any` here is the return value, we don't use it.

Reply:
Hmm... interesting. In TypeScript we use `void` as the return type for function arguments where the return value isn't used because any `(arg: string) => T` can be assigned to `(arg: string) => void`. This has an advantage over `any` because the type checker will complain if the function _does_ try to use the argument. It seems that the same approach doesn't work here though if using `Callable[[str], None]`.

This fails to typecheck:

```py
from typing import Callable

def do_something(callback: Callable[[str], None]) -> None:
    callback("foo")

def bar(arg: str) -> str:
    return arg + arg

do_something(bar)
```

This does work, but requires boilerplate:

```py
from typing import Callable, TypeVar

R = TypeVar("R", covariant=True)

def do_something(callback: Callable[[str], R]) -> None:
    callback("foo")

def bar(arg: str) -> str:
    return arg + arg

do_something(bar)
```

To be clear, this is just me being curious. I think `Any` is fine for this PR to get us started.